test/integration/noderesourcetopology_test.go

```diff
@@ -555,6 +559,254 @@ func TestTopologyMatchPlugin(t *testing.T) {
 			},
 			expectedNodes: []string{"fake-node-1", "fake-node-2"},
 		},
+		{
+			name: "SingleNUMANodePodLevel: Filtering out nodes that cannot fit resources in case of Guaranteed pod with multi containers",
+			pods: []*v1.Pod{
+				withReqAndLimit(
+					withReqAndLimit(
+						st.MakePod().Namespace(ns).Name("topology-aware-scheduler-pod"),
+						map[v1.ResourceName]string{v1.ResourceCPU: "2", v1.ResourceMemory: "4Gi"}),
+					map[v1.ResourceName]string{v1.ResourceCPU: "2", v1.ResourceMemory: "4Gi"}).Obj(),
+			},
+			nodeResourceTopologies: []*topologyv1alpha1.NodeResourceTopology{
+				makeNodeResourceTopology("fake-node-1", topologyv1alpha1.SingleNUMANodePodLevel,
+					[]topologyv1alpha1.ResourceInfoList{
+						{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "2", "2"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						},
+						{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "2", "2"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						},
+					}),
+				makeNodeResourceTopology("fake-node-2", topologyv1alpha1.SingleNUMANodePodLevel,
+					[]topologyv1alpha1.ResourceInfoList{
+						{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "4", "4"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						},
+						{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "4", "4"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "8Gi", "8Gi"),
+						},
+					}),
+			},
+			expectedNodes: []string{"fake-node-2"},
+		},
+		// the following tests are different from the NodeResourceFit tests in the sense that
+		// they would pass the NodeResourceFit since there are enough resources for the pod to be fit
+		// at the node level, but not at the node level
+		{
+			name: "SingleNUMANodeContainerLevel: Filtering out nodes that cannot fit resources in case of Guaranteed pod with multi containers",
+			pods: []*v1.Pod{
+				withReqAndLimit(
+					withReqAndLimit(
+						st.MakePod().Namespace(ns).Name("topology-aware-scheduler-pod"),
+						map[v1.ResourceName]string{v1.ResourceCPU: "3", v1.ResourceMemory: "5Gi"}),
+					map[v1.ResourceName]string{v1.ResourceCPU: "3", v1.ResourceMemory: "5Gi"}).Obj(),
+			},
+			nodeResourceTopologies: []*topologyv1alpha1.NodeResourceTopology{
+				makeNodeResourceTopology("fake-node-1", topologyv1alpha1.SingleNUMANodeContainerLevel,
```

Comment:
My take on this is that when refactoring code in a certain scope (test/integration/noderesourcetopology_test.go here) using helper functions/methods is favorable rather than using repetitive inline code for improving code readability (which is what you are doing here and is great :) ) but it is important to have a holistic view and ensure there is consistency in the code.

We shouldn't loose sight of how the file/component would look like when a new contributor starts looking at it, They shouldn't be left puzzled because of different implementation styles e.g. certain tests implemented with helper functions and others without them.

Reply:
Done. Thank you :)